tests/Doctrine/Tests/KeyValueStore/Storage/MongoDbStorageTest.php

```diff
@@ -20,40 +20,44 @@
 
 namespace Doctrine\Tests\KeyValueStore\Storage;
 
+use Doctrine\KeyValueStore\NotFoundException;
 use Doctrine\KeyValueStore\Storage\MongoDbStorage;
+use MongoDB\Client;
 
 /**
  * MongoDb storage testcase
  *
  * @author Markus Bachmann <markus.bachmann@bachi.biz>
  *
- * @requires extension mongo
+ * @covers \Doctrine\KeyValueStore\Storage\MongoDbStorage
+ * @requires extension mongodb
  */
 class MongoDbStorageTest extends \PHPUnit_Framework_TestCase
 {
-    protected function setUp()
-    {
-        $this->mongo = $this
-            ->getMockBuilder('\Mongo')
-            ->disableOriginalConstructor()
-            ->getMock();
-
-        $this->mongodb = $this->getMockBuilder('\MongoDB')->disableOriginalConstructor()->getMock();
-
-        $this->mongo->expects($this->any())
-                    ->method('selectDB')
-                    ->will($this->returnValue($this->mongodb));
+    /**
+     * @var Client
+     */
+    private $client;
 
-        $this->collection = $this->getMockBuilder('MongoCollection')->disableOriginalConstructor()->getMock();
+    /**
+     * @var MongoDbStorage
+     */
+    private $storage;
 
-        $this->mongodb->expects($this->once())
-             ->method('selectCollection')
-             ->will($this->returnValue($this->collection));
-
-        $this->storage = new MongoDbStorage($this->mongo, [
-            'collection' => 'test',
-            'database'   => 'test',
-        ]);
+    protected function setUp()
+    {
+        $this->client = new Client(
+            'mongodb://127.0.0.1/',
+            [],
+            [
+                'typeMap' => [
```

Comment:
Strictly speaking, this is unnecessary since you update the `typeMap` for the `findOne` operation in the storage class itself. Instead, we may want to leave the default type map untouched to ensure that it's being changed for an all-array type map in the storage class.